docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+* `\` escapes any of the operators for the literal character
+* `^` matches only the beginning of the string
+* `$` matches only the end of the string
+* `.` matches any single character
+* `[ ... ]` defines a character list, where the character list can also be character range. This matches any string containing these characters
+* `[^ ... ]` defines a character list, but negates them. This matches any string *not* containing these characters
+* `|` matches either expression `e1` or `e2`
+* `( ... )` groups expressions together where `...` is some regular expression
+* `*` matches the preceding character 0 or many times
+* `+` matches the preceding character at least once
+
+#### 5.1.6. `bool`
+
+Boolean objects contain a value of either `true` or `false`. They can be declared on their own, and are used in conditional statements.
+
+The structure of a boolean declaration:
+
+```
+bool switch = true; // or false
+```
+
+Example: 
+
+```
+int x = 0;
+bool switch = true;
+
+// infinite loop
+while(switch) {
+	x = x + 1;
+}
+```
+
+## 6: Statements 
+
+#### 6.1 Print Statement
+
+The print statement prints a literal value, or the value returned by an expression.
+
+`print("i will be printed to stdout");`
+
+#### 6.2 Conditional Statements
+
+Conditional statements evaluate expressions and execute code based on the truth values of those expressions.
+
+```
+if (<expression>) {
+	<code block>
+}
+elif {
+	<code block>
+}
+else {
+	<code block>
+}
+```
+
+#### 6.3 For Statements
+
+For statements iterate over an `array` and execute a code block for every iteration. The code block can mutate the array elements, but can not add or remove elements from the array.
+ex:
+
+```
+for(str current : stringArray) {
+    print(current);
+}
+``` 
+
+#### 6.4 While Statements 
+
+While statements execute a code block  until its provided condition fails to be met.
+
+```
+while(<condition>) {
+	<code block>
+}
+```
+
+### 7: Code Sample
+
+The below is an example of `FIRE` in action. In the snippet below, a `FIRE` program is used to extract phone numbers that begin with a particular area code:
+
+
+```
+user:~ $ cat PhoneNumbers.csv
+Dennis,201-445-9372
+Kenneth,954-667-8990
+Richie,312-421-0098
+Thomas,201-750-0911
+Albert,783-444-7862
+
+user:~ $ cat nj_numbers.fire
+//
+// Program that determines if a number is from NJ based on 201 area code
+//
+
+func string isNJ = (str phoneNumber) => {
+    return phoneNumber === r'201-[0-9][0-9][0-9]-[0-9][0-9][0-9][0-9]';
+};
+
+func array[int, str] extractRegion(func isRegion, file f) {
+    
+    array njnums[int, str];
+    
+    str number = f.readLine();
+    int i = 0;
+    
+    while(number != "") {
+        if(isNJ(number)){
+            njnums[i,number].add;
+	         i = i + 1;
+	         number = f.readLine();
+        }
+    }
+    return njnums;
+}
+
+file f = file("PhoneNumbers.csv", "rw", ",");
+
+print( extractRegion(isNJ, f) );
+
+
+user:~ $ cut -d' ' -f2 | fire nj_numbers.fire
```

Comment:
how this is run doesn't much sense...

1. your language isn't interpreted. you're building a compiler, so invoking `fire` should just compile the `.fire` code, but not actually execute it.

2. why is `cut` being piped into `fire`? how does your language handle `stdin`? and what file is `cut ` operating on?

Reply:
edited